test/expr/print/misc_expr.swift

```diff
@@ -0,0 +1,117 @@
+// RUN: %target-swift-frontend -print-ast %s 2>&1 | %FileCheck %s
+
+var x = 0.0
+
+type(of: x)
+// CHECK: type(of: x)
+
+class A {
+  var x: Int = 3
+  init() {
+  }
+}
+
+class B: A {
+  override init() {
+    super.init()
+    super.x = 2;
+  }
+}
+// CHECK: @_inheritsConvenienceInitializers internal class B : A {
+// CHECK:   override internal init() {
+// CHECK:     super.init()
+// CHECK:     super.x = 2
+// CHECK:   }
+// CHECK:   @objc deinit {
+// CHECK:   }
+// CHECK: }
+
+var a: [Int] = [1, 3, 2];
+
+for i in 0...2 {
+  a[i] = i;
+}
+// CHECK: for i in 0 ... 2 {
+// CHECK:   a[i] = i
+// CHECK: }
+
+let y: Double = 0 as Double
+// CHECK: @_hasInitialValue internal let y: Double = 0 as Double
+
+var stringToInt: Int? = Int("1");
+
+var forceInt: Int = stringToInt!
+// CHECK: @_hasInitialValue internal var forceInt: Int = stringToInt!
+
+var prefixUnaryExpr: Int = -a[1];
+// CHECK: @_hasInitialValue internal var prefixUnaryExpr: Int = -a[1]
+var postfixUnaryExpr: Int = stringToInt!
+// CHECK: @_hasInitialValue internal var postfixUnaryExpr: Int = stringToInt!
+
+class MyID {
+  var num = 1
+}
+
+class SomeJSON {
+  var id: MyID?
+}
+
+let cnt = SomeJSON().id?.num
+// CHECK: @_hasInitialValue internal let cnt: Int? = SomeJSON().id?.num
+
+struct User {
+  let name: String
+  let email: String
+  let address: Address?
+  let role: Role
+}
+
+struct Address {
+  let street: String
+}
+
+enum Role {
+  case admin
+  case member
+  case guest
+  
+  var permissions: [Permission] {
+    switch self {
+    case .admin:
+        return [.create, .read, .update, .delete]
+    case .member:
+        return [.create, .read]
+    case .guest:
+        return [.read]
+    }
+  }
+}
+
+enum Permission {
+  case create
+  case read
+  case update
+  case delete
+}
+
+var user = User(
+  name: "Swift",
+  email: "http://www.swift.org",
+  address: nil,
+  role: .admin
+)
+
+
+let userRoleKeyPath = \User.role
+// CHECK: @_hasInitialValue internal let userRoleKeyPath: KeyPath<User, Role> = \User
+
+let role = user[keyPath: userRoleKeyPath]
+// CHECK: @_hasInitialValue internal let role: Role = user[keyPath: userRoleKeyPath]
+
+struct FakeColor: _ExpressibleByColorLiteral {
+  init(_colorLiteralRed: Float, green: Float, blue: Float, alpha: Float) {}
+}
+typealias _ColorLiteralType = FakeColor
+
+let myColor = #colorLiteral(red: 0.292, green: 0.081, blue: 0.6, alpha: 255)
+// CHECK: @_hasInitialValue internal let myColor: FakeColor = colorLiteral(red: 0.292, green: 0.081, blue: 0.6, alpha: 255)
```

Comment:
I think it would be great if we could change these tests so they also do a typecheck of the output to ensure it's valid Swift, as that would help avoid these kinds of issues. You should be able to write it as something like:

```
// RUN: %empty-directory(%t)
// RUN: %target-swift-frontend -print-ast %s > %t/result.swift
// RUN: %target-typecheck-verify-swift %t/result.swift
// RUN: %FileCheck %s < %t/result.swift
```

Reply:
Do you mean to add this for every `test` file w.r.t to `ASTPrinter`?

Reply:
Ideally yes, there may be some constructs that it won't work for, e.g memberwise initializers since they have their bodies generated by SILGen, but we could extract that testing out to a separate file that doesn't do a typecheck of the result.